File: src/main/java/com/dh/ondot/core/config/WebConfig.java
package com.dh.ondot.core.config;

import com.dh.ondot.core.interceptor.TokenInterceptor;
import com.dh.ondot.member.core.OauthProviderConverter;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Configuration;
import org.springframework.format.FormatterRegistry;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
@RequiredArgsConstructor
public class WebConfig implements WebMvcConfigurer {
    private final OauthProviderConverter oauthProviderConverter;
    private final TokenInterceptor tokenInterceptor;

    @Override
    public void addCorsMappings(CorsRegistry registry) {
        registry.addMapping("/**")
                .allowedOriginPatterns("*")
                .allowedMethods("GET", "POST", "PUT", "DELETE", "PATCH", "OPTIONS")
                .allowedHeaders("*")
                .maxAge(3600);
    }

    @Override
    public void addFormatters(FormatterRegistry registry) {
        registry.addConverter(oauthProviderConverter);
    }

    @Override
    public void addInterceptors(InterceptorRegistry registry) {
        registry.addInterceptor(tokenInterceptor)
                .addPathPatterns("/members/**", "/alarms/**", "/places/**", "/schedules/**");
    }
}

File: src/main/kotlin/com/dh/ondot/core/config/WebConfig.kt
package com.dh.ondot.core.config

import com.dh.ondot.core.TokenInterceptor
import com.dh.ondot.member.core.OauthProviderConverter
import org.springframework.context.annotation.Configuration
import org.springframework.format.FormatterRegistry
import org.springframework.web.servlet.config.annotation.CorsRegistry
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class WebConfig(
    private val oauthProviderConverter: OauthProviderConverter,
    private val tokenInterceptor: TokenInterceptor,
) : WebMvcConfigurer {

    override fun addCorsMappings(registry: CorsRegistry) {
        registry.addMapping("/**")
            .allowedOriginPatterns("*")
            .allowedMethods("GET", "POST", "PUT", "DELETE", "PATCH", "OPTIONS")
            .allowedHeaders("*")
            .maxAge(3600)
    }

    override fun addFormatters(registry: FormatterRegistry) {
        registry.addConverter(oauthProviderConverter)
    }

    override fun addInterceptors(registry: InterceptorRegistry) {
        registry.addInterceptor(tokenInterceptor)
            .addPathPatterns("/members/**", "/alarms/**", "/places/**", "/schedules/**")
            .excludePathPatterns(
                "/schedules/*/issues",
                "/schedules/*/preparation",
            )
    }
}
